Customer home: add section

eliorivero opened the pull request:
#### Changes proposed in this Pull Request

* introduces the section for the upcoming Customer Home

<img width="1164" alt="Screen Shot 2019-08-07 at 18 41 30" src="https://user-images.githubusercontent.com/1041600/62660026-55815f00-b943-11e9-8ce3-9580f817bf3d.png">

* if the user can't see the Customer Home, they’ll see a message

<img width="1109" alt="Screen Shot 2019-08-08 at 14 38 58" src="https://user-images.githubusercontent.com/1041600/62726490-b6657180-b9ed-11e9-89c8-9c301898a06c.png">

#### Testing instructions

* manually type `/home/%site slug%` and `/home`
* both paths should work
* links should go to the correct section
* ensure it's only visible on development. Staging and wpcalypso shouldn't display this.

eliorivero committed:
Customer home: add section

eliorivero committed:
Use customer-home config flag to show Checklist menu item in production, but Customer Home in development

eliorivero committed:
Initial content for Customer Home section

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+							</div>
+							<ul className="customer-home__card-boxes">
+								<li>{ translate( 'Add a page' ) }</li>
+								<li>{ translate( 'Write blog post' ) }</li>
```
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Write a Blog Post' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+							<ul className="customer-home__card-boxes">
+								<li>{ translate( 'Add a page' ) }</li>
+								<li>{ translate( 'Write blog post' ) }</li>
+								<li>{ translate( 'Customize theme' ) }</li>
```
Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Customize Theme' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+								<li>{ translate( 'Write blog post' ) }</li>
+								<li>{ translate( 'Customize theme' ) }</li>
+								<li>{ translate( 'Change theme' ) }</li>
+								<li>{ translate( 'Edit menus' ) }</li>
```
Hi! I've found a possible matching string that has already been translated **127** times:
`translate( 'Edit Menus' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+							</h6>
+							<ul className="customer-home__card-links">
+								<li>{ translate( 'Share my site' ) }</li>
+								<li>{ translate( 'Grow my audience' ) }</li>
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Yes! I want to grow my audience.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+							<ul className="customer-home__card-links">
+								<li>{ translate( 'Share my site' ) }</li>
+								<li>{ translate( 'Grow my audience' ) }</li>
+								<li>{ translate( 'Earn money' ) }</li>
```
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Earn Money' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+							</h6>
+							<div className="customer-home__card-button-pair">
+								<Button href="https://play.google.com/store/apps/details?id=org.wordpress.android">
+									{ translate( 'Google Play' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Get on Google Play' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,121 @@
+									{ translate( 'Google Play' ) }
+								</Button>
+								<Button href="https://apps.apple.com/us/app/wordpress/id335703880">
+									{ translate( 'Apple App Store' ) }
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Apple App Store download badge' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero committed:
Finish adding content & styles

eliorivero committed:
Remove flags from staging and wpcalypso

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+								/>
+								<ActionBox
+									action={ () => page( `/post/${ siteSlug }` ) }
+									label={ translate( 'Write blog post' ) }
```
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Write a Blog Post' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+								/>
+								<ActionBox
+									action={ customizeUrl }
+									label={ translate( 'Customize theme' ) }
```
Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Customize Theme' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+								/>
+								<ActionBox
+									action={ customizeUrl }
+									label={ translate( 'Edit menus' ) }
```
Hi! I've found a possible matching string that has already been translated **127** times:
`translate( 'Edit Menus' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+									{ translate( 'Share my site' ) }
+								</VerticalNavItem>
+								<VerticalNavItem path={ `/marketing/tools/${ siteSlug }` }>
+									{ translate( 'Grow my audience' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Yes! I want to grow my audience.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+									{ translate( 'Grow my audience' ) }
+								</VerticalNavItem>
+								<VerticalNavItem path={ `/earn/${ siteSlug }` }>
+									{ translate( 'Earn money' ) }
```
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Earn Money' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,192 @@
+							<div className="customer-home__card-button-pair customer-home__card-mobile">
+								<Button
+									href="https://play.google.com/store/apps/details?id=org.wordpress.android"
+									aria-label={ translate( 'Google Play' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Get on Google Play' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos committed:
Responsive style tweaks, to drop the layout to a single column

pablinos commented on client/my-sites/customer-home/index.js:
```diff
@@ -0,0 +1,18 @@
+export default function() {
+	page( '/home', siteSelection, sites, makeLayout, clientRender );
+
+	page( '/home/:site_id', siteSelection, navigation, home, makeLayout, clientRender );
```
I know this is primarily focused on layout at the moment, but should we add a redirect here if `canCurrentUserUseCustomerHome` is false?

eliorivero replied:
As per our discussion, will add a message in the section itself, informing the user about this.

eliorivero committed:
If user can't see Customer Home, display a message informing about it

eliorivero committed:
Merge branch 'add/customer-home-section' of github.com:Automattic/wp-calypso into add/customer-home-section

eliorivero commented:
In https://github.com/Automattic/wp-calypso/pull/35169/commits/1531a59058c8e2de773f3a098ccf69028fe6310e

If the user can't see Customer Home, display a message informing about it

<img width="1109" alt="Screen Shot 2019-08-08 at 14 38 58" src="https://user-images.githubusercontent.com/1041600/62726490-b6657180-b9ed-11e9-89c8-9c301898a06c.png">

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									iconSrc="/calypso/images/customer-home/menus.svg"
+								/>
+								<ActionBox
+									action="https://en.support.wordpress.com/images/"
```
I don't think it's necessary to force `en` here

```suggestion
									action="https://support.wordpress.com/images/"
```

eliorivero replied:
Yeah I think it's fine so `es` and `br` can display the page to go to the original english content. Thank you!

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									iconSrc="/calypso/images/customer-home/logo.svg"
+								/>
+								<ActionBox
+									action="https://wordpress.com/email/domain.wordpress.com"
```
I'm assuming this is a placeholder which needs updating. :)

eliorivero replied:
Thanks, good catch, fixed 👍

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									alt={ translate( 'Support' ) }
+								/>
+								<VerticalNav className="customer-home__card-links">
+									<VerticalNavItem path="https://en.support.wordpress.com/" external>
```
```suggestion
									<VerticalNavItem path="https://support.wordpress.com/" external>
```

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									<VerticalNavItem path="https://en.support.wordpress.com/" external>
+										{ translate( 'Support docs' ) }
+									</VerticalNavItem>
+									<VerticalNavItem path="https://wordpress.com/help/contact" external>
```
I think that this would be helpful to keep the flows when not on WordPress.com

```suggestion
									<VerticalNavItem path="/help/contact" external>
```

Aurorum commented on client/my-sites/customer-home/style.scss:
```diff
@@ -0,0 +1,115 @@
+}
+
+.customer-home__card-subheader {
+	color: var( --color-neutral-light );
```
This doesn't pass contrast requirements.

```suggestion
	color: var( --color-text-subtle );
```

eliorivero replied:
@davemart-in what do you think?

#### Current, `--color-neutral-light`

<img width="497" alt="Screen Shot 2019-08-09 at 19 59 56" src="https://user-images.githubusercontent.com/1041600/62813457-37e9fc00-bae1-11e9-9855-acc7a726fe2c.png">

#### Suggested, `--color-text-subtle`

<img width="496" alt="Screen Shot 2019-08-09 at 19 59 42" src="https://user-images.githubusercontent.com/1041600/62813458-37e9fc00-bae1-11e9-8d99-2dd25aa15a7d.png">

davemart-in replied:
I like the contrast boost!

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									href="https://play.google.com/store/apps/details?id=org.wordpress.android"
+									aria-label={ translate( 'Google Play' ) }
+								>
+									<img src="/calypso/images/customer-home/google-play.png" alt="" />
```
Hmm, does it make sense to add this image to the repo again? It already exists here: https://github.com/Automattic/wp-calypso/blob/master/static/images/me/get-apps-google-play.png

eliorivero replied:
It has a border, so we'll keep the new one.

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,214 @@
+									href="https://apps.apple.com/us/app/wordpress/id335703880"
+									aria-label={ translate( 'App Store' ) }
+								>
+									<img src="/calypso/images/customer-home/apple-store.png" alt="" />
```
Same as above, but it lives here! 😄 
https://github.com/Automattic/wp-calypso/blob/master/static/images/me/get-apps-ios-store.svg

eliorivero replied:
It has a border, so we'll keep the new one.

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,212 @@
+		if ( ! canUserUseCustomerHome ) {
+			return (
+				<EmptyContent
+					title={ preventWidows( translate( 'You are not authorized to view this page.' ) ) }
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero committed:
Add checklist (wpcom and jetpack) to customer home

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+							/>
+							<ActionBox
+								action={ () => page( `/post/${ siteSlug }` ) }
+								label={ translate( 'Write blog post' ) }
```
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Write a Blog Post' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+							/>
+							<ActionBox
+								action={ customizeUrl }
+								label={ translate( 'Customize theme' ) }
```
Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Customize Theme' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+							/>
+							<ActionBox
+								action={ customizeUrl }
+								label={ translate( 'Edit menus' ) }
```
Hi! I've found a possible matching string that has already been translated **127** times:
`translate( 'Edit Menus' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/customer-home-section)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+								{ translate( 'Share my site' ) }
+							</VerticalNavItem>
+							<VerticalNavItem path={ `/marketing/tools/${ siteSlug }` }>
+								{ translate( 'Grow my audience' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Yes! I want to grow my audience.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+								{ translate( 'Grow my audience' ) }
+							</VerticalNavItem>
+							<VerticalNavItem path={ `/earn/${ siteSlug }` }>
+								{ translate( 'Earn money' ) }
```
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Earn Money' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+						<div className="customer-home__card-button-pair customer-home__card-mobile">
+							<Button
+								href="https://play.google.com/store/apps/details?id=org.wordpress.android"
+								aria-label={ translate( 'Google Play' ) }
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Get on Google Play' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero committed:
Update links for support and contact. Props @Aurorum.

Remove the en in the link to the images support doc so it takes users first to the en and br versions. This will help when user uses a spanish or portuguese UI so they're not surprised when they're taken to an english page.

ramonjd commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+				<SidebarNavigation />
+				{ siteId && isEmpty( siteChecklist ) && <QuerySiteChecklist siteId={ siteId } /> }
+				{ siteId &&
+					! isEmpty( siteChecklist ) &&
```
Since you're checking `isEmpty( siteChecklist )` twice, maybe you could assign that value to a variable. 

Also, it looks like we're importing two lodash methods, `isEmpty` and `some` for operations we could do with plain JS, especially since you're already assigning `{}` to `siteChecklist` in the connect method.

```
const  hasChecklistTasks = siteChecklist && Array. isArray( siteChecklist.tasks );

....

				{ siteId && ! hasChecklistTasks && <QuerySiteChecklist siteId={ siteId } /> }
                { siteId
                    && hasChecklistTasks
                    && siteChecklist.tasks.some( task => task. isCompleted === false )
                    ? checklistToRender
                    : this.renderCustomerHome()
                }  
```

Alternatively, you could pass this stuff as props via connect. Might make unit testing the UI easier too.

```
    const hasChecklistTasks: siteChecklist !== null && Array. isArray( siteChecklist.tasks )
	return {
       ....
        hasChecklistTasks,
        shouldRenderChecklist: siteId && hasChecklistTasks && siteChecklist.tasks.some( task => task. isCompleted === false ),
	};
```

As always, feel free to ignore this! :) Looking forward to seeing the home live in production.

eliorivero replied:
Thanks for the suggestion! I added props to connect, although with a slight change. Didn't check for siteId because I realized it's enough to know if siteChecklist is null. I also removed the siteId check in the render method for the same reason, specially now that shouldRenderChecklist encompasses everything.
Thanks again!
```js
shouldRenderChecklist: hasChecklistTasks && ...
```

ramonjd commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,254 @@
+import './style.scss';
+
+const ActionBox = ( { action, iconSrc, label } ) => {
+	const buttonAction = 'function' === typeof action ? { onClick: action } : { href: action };
```
I don't know if this is any better, but we could make this function a bit 'purer' by not mixing types on `action`

```
const ActionBox = ( { href, iconSrc, label, onClick } ) => {
	const buttonProps = { href, onClick };
    return (
		<div className="customer-home__box-action">
        <Button { ...buttonProps }>
       .....


							<ActionBox
								onClick={ () => page( `/themes/${ siteSlug }` ) }
								label={ translate( 'Change theme' ) }
								iconSrc="/calypso/images/customer-home/theme.svg"
							/>
							<ActionBox
								href={ customizeUrl }
								label={ translate( 'Edit menus' ) }
								iconSrc="/calypso/images/customer-home/menus.svg"
							/>

```

eliorivero replied:
Nice suggestion, thanks

ramonjd commented on client/my-sites/customer-home/index.js:
```diff
@@ -0,0 +1,18 @@
+export default function() {
+	page( '/home', siteSelection, sites, makeLayout, clientRender );
+
+	page( '/home/:site_id', siteSelection, navigation, home, makeLayout, clientRender );
```
Does this need snake case for any reason here and not `siteId`?

Not a biggie, just Calypso convention

eliorivero replied:
Not really sure where it came from but I've changed it now. Thanks!

pablinos committed:
Remove the checklist from the plans page for Jetpack sites

Fixes #35288

eliorivero committed:
Increase contrast so it passes accessibility requirements

Co-Authored-By: Aurorum <torres126@clubpenguinmountains.com>

eliorivero committed:
Update logic to show the Checklist. Props @ramonjd

eliorivero committed:
We like camels not snakes.

eliorivero committed:
Avoid unnecessary condition and pass both props to Button component

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,258 @@
+						</h6>
+						<div className="customer-home__card-support">
+							<img
+								src="/calypso/images/customer-home/happiness.png"
```
This image seems a bit odd for three reason:

- It seems really blurry compared to other images in Calypso
- The colours vary from the rest of the ones in Calypso (#30248)

Those could be intentional, but this feels like a bug:
- The proportions are off on my tablet

![Screenshot_20190812-220957](https://user-images.githubusercontent.com/43215253/62898781-3bce8600-bd4e-11e9-99ac-7b611c963e67.jpg)

eliorivero replied:
Interesting! Can you provide more info about the device where you're experiencing this stretching?

Aurorum replied:
Samsung Galaxy Tab S2 (Android 7) :)

Aurorum commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,258 @@
+								iconSrc="/calypso/images/customer-home/logo.svg"
+							/>
+							<ActionBox
+								href={ `https://wordpress.com/email/${ siteSlug }` }
```
Just to keep the flows again. :) 

```suggestion
								href={ `/email/${ siteSlug }` }
```

eliorivero replied:
Thank you. It has to be `onClick={ () => page( `/email/${ siteSlug }` ) }` though since we'll add something else later, but thank you very much for noticing this.

Aurorum commented on client/state/sites/selectors/can-current-user-use-customer-home.js:
```diff
@@ -0,0 +1,21 @@
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether user can access the Earn section.
```
Assuming this is unintentional :) 

```suggestion
 * @return {?Boolean}        Whether user can access the CustomerHome section.
```

eliorivero replied:
Right. Although it should be Customer*space*Home. It looks like when I did a massive update I replaced Earn with CustomerHome and it ended up like that. Thanks for catching this.

eliorivero committed:
Use event instead of url to later add tracking. Amend inline phpdoc.

pablinos commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,258 @@
+				<DocumentHead title={ translate( 'Customer Home' ) } />
+				<SidebarNavigation />
+				{ siteId && ! hasChecklistData && <QuerySiteChecklist siteId={ siteId } /> }
+				{ shouldRenderChecklist ? this.renderChecklist() : this.renderCustomerHome() }
```
I'm not sure if this is the best way to do it, but this uses `hasChecklistData` to stop the Customer  Home displaying while the data is fetched.

```suggestion
				{ shouldRenderChecklist && this.renderChecklist() }
				{ hasChecklistData && ! shouldRenderChecklist && this.renderCustomerHome() }
```

pablinos committed:
Update logic to prevent rendering while the checklist data is fetched

eliorivero committed:
Introduce isSiteChecklistComplete. Use it to decide when to render the checklist. Don't render anything if it's null.

pablinos commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -86,16 +92,19 @@ class Home extends Component {
 				{ siteId && ! hasChecklistData && <QuerySiteChecklist siteId={ siteId } /> }
-				{ hasChecklistData &&
-					( shouldRenderChecklist ? this.renderChecklist() : this.renderCustomerHome() ) }
+				{ null !== isChecklistComplete && (
```
Should we use `hasChecklistData` here? It feels more descriptive, but there might be something else in this logic.

pablinos commented on client/state/selectors/is-site-checklist-complete.js:
```diff
@@ -0,0 +1,54 @@
+		@param   {Object}  task The task that we'll check to see if it's completed.
+		@returns {Boolean}      Whether the task is considered to be completed or not.
+	*/
+	const isTaskReady = task => {
```
I don't think this matters too much as there's a great explanation, but is there a better word than `Ready`? I'm struggling to think of one. `Fulfilled`, `Concluded`, `Closed`, `Finished` - nothing seems quite right!

eliorivero replied:
Completed. Not sure why I wrote ready 😅

pablinos commented on client/state/selectors/is-site-checklist-complete.js:
```diff
@@ -0,0 +1,54 @@
+	const siteChecklist = getSiteChecklist( state, siteId );
+
+	if (
+		isSiteChecklistLoading( state, siteId ) ||
```
I had to remove this line to prevent a loop where the data is constantly requested. I think the render kept happening before the network request completed. Swapping to using `hasChecklistData` in the main component would probably do it too.

pablinos replied:
This was confusing me and the problem comes from here I think https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checklist/main.jsx#L194 The `QuerySiteChecklist` component is rendered regardless of whether the data is available, but then that means that `isSiteChecklistLoading` keeps returning `true` and we get stuck in a loop. 

I feel like there should be a better pattern for retrieving this data 🤔

eliorivero replied:
I noticed–even before embedding the Checklist in the CH–that the Checklist constantly sends requests. They're doing that on purpose I guess, maybe to be constantly get results like which task was just completed? In any case, I think it's beyond the scope of this PR.
And I think `isSiteChecklistLoading` does return false briefly. Enough time to render the checklist.

eliorivero committed:
Change "ready" expressions to "completed" since it's more in line with Checklist

pablinos committed:
Use `hasChecklistData` to prevent request loop

We were getting in to a loop requesting the checklist data because the
checklist component had another instance of the `QuerySiteChecklist`
component, which meant the data seemed to be constantly in a loading
state.

We should probably refactor this to remove one of the query components.

pablinos committed:
Merge branch 'add/customer-home-section' of github.com:Automattic/wp-calypso into add/customer-home-section

pablinos committed:
Prevent the display of the customer home to Jetpack sites

pablinos committed:
Add logic to distinguish between unauthorized users and ineligible sites

The customer home screen is not available for Jetpack sites and will be
A/B tested, so this adds a selector to determine if a site is eligible
for the new customer home screen. This means we can tailor the messaging
on the error screen if a user navigates to the home screen on a site
where it's not available.

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,275 @@
+
+		if ( ! canUserUseCustomerHome ) {
+			const title = this.props.isSiteEligible
+				? translate( 'You are not authorized to view this page.' )
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -73,15 +74,19 @@ class Home extends Component {
 			}
 		},
 		isJetpackNotAtomic: PropTypes.bool.isRequired,
+		isSiteEligible: PropTypes.bool.isRequirod,
```
Typo here, `isRequirod`. Will fix.

eliorivero committed:
Remove all Jetpack related stuff. Fix typo isRequirod that was throwing warnings.

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,253 @@
+
+		if ( ! canUserUseCustomerHome ) {
+			const title = this.props.isSiteEligible
+				? translate( 'You are not authorized to view this page.' )
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos committed:
Remove the body text from the error page

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,248 @@
+
+		if ( ! canUserUseCustomerHome ) {
+			const title = this.props.isSiteEligible
+				? translate( 'You are not authorized to view this page.' )
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero committed:
Fix front page id fetching

eliorivero committed:
Display dismissable banner when the checklist is completed in the Customer Home

eliorivero committed:
Display dismissable banner when the checklist is completed in the Customer Home

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,288 @@
+
+		if ( ! canUserUseCustomerHome ) {
+			const title = this.props.isSiteEligible
+				? translate( 'You are not authorized to view this page.' )
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -69,6 +70,35 @@ class Home extends Component {
+			null === nextProps.isChecklistComplete ||
+			'norender' === prevState.renderChecklistCompleteBanner
+		) {
+			return null;
```
I was getting a warning about the component `state` being undefined. Should we return `prevState` here?

eliorivero replied:
Did something changed in React? Last I remember was that state was returning null in getderivedstatefromprops was like saying that the state didn't change.
Will address this.

pablinos commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -84,25 +114,35 @@ class Home extends Component {
-					</Fragment>
+				{ renderChecklistCompleteBanner && (
+					<Banner
+						dismissPreferenceName
```
I was getting another warning about this not being the right type. I changed it to:

```
dismissPreperenceName="checklist-complete"
```

and it seemed satisfied.

eliorivero replied:
Cool, will fix

eliorivero committed:
Merge branch 'add/customer-home-section' of github.com:Automattic/wp-calypso into add/customer-home-section

eliorivero committed:
Fix issues with getDerived... returning null so it doesn't complain about an undefined state and the correct type in a Banner prop

a8ci18n commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -0,0 +1,289 @@
+
+		if ( ! canUserUseCustomerHome ) {
+			const title = this.props.isSiteEligible
+				? translate( 'You are not authorized to view this page.' )
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'You are not authorized to view this page' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

eliorivero committed:
Fix distorted image on Samsung Galaxy Tab S2/S3, fixes #35416
